### ISSUE TITLE
Support for YAML struct tags instead of JSON.

### ISSUE THREAD

**Issue by liamkinne** [deepmap/oapi-codegen#849]
For my use case I'd like the struct tags to be for YAML instead of JSON. I've tried 'applicaion/yaml' in the response type of the API spec had a look at the code and it doesn't seem this project supports that out of the box.

What I get
```go
type Device struct {
	Messages *[]Message `json:"messages,omitempty"`
}
```

What I want
```go
type Device struct {
	Messages *[]Message `yaml:"messages,omitempty"`
}
```

If this isn't already a feature, or on the road map I am happy to fork and give implementing this a crack. Just wanted to get a second pair of eyes to confirm.

**Comment by liamkinne** [comment]
For anyone that comes across this and wants a quick fix. `sed` to the rescue.

```shell
sed -i '' "s/json:/yaml:/" api.gen.go
```

**Comment by jamietanna** [comment]
To confirm, is this where you're expecting to produce an `application/yaml` formatted API response?

**Comment by liamkinne** [comment]
@jamietanna yes, produce (with server generation) or consume (with client generation) yaml formatted data.

**Comment by deitch** [comment]
I would like to see this as well. I don't mind attempting a PR for it, assuming it will get moved through. Based on the comments on [`GenFieldsFromProperties()`](https://github.com/deepmap/oapi-codegen/blob/master/pkg/codegen/schema.go#L661-L757), the generation happens in that func, and specifically [these lines](https://github.com/deepmap/oapi-codegen/blob/master/pkg/codegen/schema.go#L720-L730).

It shouldn't be too hard to add a similar section for yaml, with `p.yamlFieldName` (which might or might not be identical to `p.jsonFieldName`. I would have to dig a bit to see where it is set.

Interested?

**Comment by deitch** [comment]
> It shouldn't be too hard to add a similar section for yaml, with p.yamlFieldName (which might or might not be identical to p.jsonFieldName

Actually, I overthought this. By definition, `jsonFieldName` must be the same as the field name for yaml, as it is the field from the original spec that generated this. So this should be easier than I thought, just adding a few lines there.

@jamietanna want a short PR for it?

**Comment by mromaszewicz** [comment]
The requested feature (YAML struct tags) was implemented in PR #1798 (merged Nov 2024), which closed the related issue #1696. The feature is available via the output-option 'yaml-tags: true' in the codegen config, which causes both json: and yaml: struct tags to be generated. An integration test in internal/test/outputoptions/yaml-tags/ confirms the feature works as expected.

Closing as fixed. If you can still reproduce on a recent release, please reopen with the exact spec and version used.
